cloud/services/securitygroups/securitygroups.go

```diff
@@ -140,6 +140,11 @@ func (s *Service) Delete(ctx context.Context) error {
 	ctx, span := tele.Tracer().Start(ctx, "securitygroups.Service.Delete")
 	defer span.End()
 
+	if !s.Scope.IsVnetManaged() {
```

Comment:
let's add a unit test case for this as well

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/cloud/services/securitygroups/securitygroups_test.go#L255

Reply:
I've added a unit-test (basically copy-pasted the one from `reconcile`) let me know what you think. Thanks.